includes/modules/learndash.php

```diff
@@ -70,15 +70,17 @@ public static function pmpro_has_membership_access( $post_id = null, $user_id =
 	 */
 	public static function has_access_to_post( $post_id = null, $user_id = null ) {
 		global $post;
-		
+
 		// Use post global or queried object if no $post_id was passed in.
-		// Copied from PMPro includes/content.php.
-		if( ! $post_id && ! empty( $post ) && ! empty( $post->ID ) ) {
-			$post_id = $post->ID;
-		} elseif( ! $post_id && ! empty( $queried_object ) && ! empty( $queried_object->ID ) ) {
-			$post_id = $queried_object->ID;
+		if( ! $post_id ){
+			$post_id = get_the_ID();
 		}
-		
+if ( ! $post_id ) {
+    $queried_object = get_queried_object;
```

Comment:
```suggestion
    $queried_object = get_queried_object();
```

Reply:
Noticed, but changed to get_queried_object_id(), check this commit https://github.com/strangerstudios/pmpro-courses/pull/18/commits/d474a3233b8e31f3cf093c0abd7df2bd4702c22f

Still from wp 3.1.0+, $post_id is always a int, no type juggling so far

Reply:
If statement needs a boolean, so `if (boolean) ...`
Provided `$post_id` is and integer `if (! integer) ...` is equal to `if (! (bool)integer) ...`
[`get_the_id()`](https://developer.wordpress.org/reference/functions/get_the_id/) returns int|false so:
`if ($post_id === false) ...`

Reply:
I would agree in general. But when it comes to WordPress and specifically when working with post IDs, they are frequently treated as strings, there's no much attention to the type in general; also, only positive numbers are valid post ids (0 is not a valid post id). This is way `! $post_id` is almost everywhere in WordPress code.

Back to this case, `get_the_id()` should return a positive int, but since it uses a global accessible `$post` variable (thanks to `get_post()`), we can't be sure that `$post->ID` is a positive int.

That's why I think that `! $post_id` is safer for us, even if not 100% correct, considering how bad developers are in general and considering how much spaghetti code exists in WordPress' plugins environment.

Reply:
> there's no much attention to the type in general

I see. I see!
You must know I'm a PHPStan extension author. Static analysis guards my way even with tiny PHP files.

Reply:
Did you know that WordPress core is being fixed using PHPStan? See Trac!

Reply:
I've never felt matching my work morale to WordPress ecosystem. We could/should write high quality code even for WordPress :)

Reply:
Thanks, everyone. Love your conversation. :)